### ISSUE TITLE
Implement effects DB processors

### ISSUE THREAD

**Issue by bartekn** [stellar/go#1981]
- While working on this please make sure to use [`BatchInsertBuilder`](https://godoc.org/github.com/stellar/go/support/db#BatchInsertBuilder). Without it processing will be too slow for large ledger (we experience this in the old ingestion).
- Should be easy to do with [`LedgerTransaction.GetChanges`](https://godoc.org/github.com/stellar/go/exp/ingest/io#LedgerTransaction.GetChanges).
- The old ingestion system has this huge [`switch`](https://github.com/stellar/go/blob/master/services/horizon/internal/ingest/session.go#L151) checking operation type and creating effects based on this. Would be great to use pipeline to pass only transactions with operations a given processor is interested. So something like:
```
-> FilterTransactionsProcessor{OperationType: CreateAccount}
  -> CreateAccountEffectsProcessor{}
-> FilterTransactionsProcessor{OperationType: Payment}
  -> PaymentEffectsProcessor{}
```
OR, filter operations needed by a given effects group:
```
-> FilterTransactionsProcessor{OperationType: []{CreateAccount, Payment, PathPayment, ...}
  -> AccountEffectsProcessor{}
-> FilterTransactionsProcessor{OperationType: []{SetOptions}
  -> SignerEffectsProcessor{}
```
both ways will make the code easier to understand and test.

**Comment by bartekn** [comment]
When working on this make sure that data is in the same order in all Horizon servers. Currently it's not the case because the order of tx meta can be different in different stellar-core servers (stored in unordered map internally). Make sure Horizon responds with exactly the same data will help tools like `horizon-cmp`.

**Comment by abuiles** [comment]
Fixed in #2104
